health-checker/src/main/java/io/stargate/health/BundleService.java

```diff
@@ -32,14 +37,12 @@ public boolean checkBundleStates() {
     return true;
   }
 
-  public boolean checkIsReady() {
-    ServiceReference dataStoreFactoryReference =
-        context.getServiceReference(DataStoreFactory.class.getName());
+  public boolean checkDataStoreAvailable() {
```

Comment:
Periodic, yes.

As for caching the factory reference, is the get/unget operation too expensive? I hope not, and going through OSGi add extra validation that the service is operational since OSGi is expected to refuse the `get` operation if the bundle is no longer valid (e.g. undeployed).

Reply:
Regarding being too expensive, I am not sure. Good point about additional validation.